aten/src/ATen/cpu/vec256/vec256_bool.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ATen/cpu/vec256/intrinsics.h>
+#include <ATen/cpu/vec256/vec256_base.h>
+
+namespace at {
+namespace vec256 {
+namespace {
+
+#ifdef __AVX2__
+
+template <>
+struct Vec256<uint8_t> {
+  static constexpr int size() {
+    return 32;
+  }
+};
+
+template <>
+Vec256<uint8_t> inline operator&&(
```

Comment:
Pushing this up for testing. I am trying to figure out how to trigger this in my local test. 

This PR is only doing this for `uint8_t` for now. If this works and does speed it up, I will add another PR to implement similar algorithm for other `vec256_*.h` files.

`Vec256<uint8_t>` is not directly using `__m256i ` as its value, as I am worried that its `blend` and `set` methods might need to use the sign bit of `int64_t`, which I think is not wrong but is a little weird. 

@fmassa @gchanan any suggestion?